### PR TITLE
#164798295 Enable a user comment on an article

### PR DESCRIPTION
**Pull request by patrickf949**
#### What does this PR do?
Ensures users comment on Articles

#### Description of Task to be completed?
- Write tests for model creation
- Create models for comments
- Create models for replies to comments
- Add Urls for commment manipulation
- Add view classes  for comment and reply manipulation

#### How should this be manually tested?
- Run the server
- Connect to swagger docs using this link
`localhost:8000/swagger-docs/`

Enter this data in the endpoint
- api/v1/articles/comments/{slug}/

```
{
	"body":"This is my comment"
}
```

Expect a result similar to this
```
{
    "comment": {
        "id": 4,
        "author": {
            "username": "patritzw",
            "bio": "",
            "email": "patrick.oakls@andeqsdla.com",
            "full_name": "",
            "image": "",
            "followers_no": 0,
            "following_no": 0
        },
        "article": {
            "author": {
                "username": "patritzw",
                "bio": "",
                "email": "patrick.oakls@andeqsdla.com",
                "full_name": "",
                "image": "",
                "followers_no": 0,
                "following_no": 0
            },
            "title": "Cedes",
            "average_rating": 0,
            "description": "You jus adsaffaawonaseat",
            "body": "Oh really, I didsassa, ifdafsds that asfawetrue",
            "createdAt": "2019-04-15T12:08:37.101772Z",
            "updatedAt": "2019-04-15T12:08:37.101791Z",
            "slug": "cedes-0x3",
            "tagList": []
        },
        "body": "Na thats some sht",
        "parentId": 0,
        "createdAt": "2019-04-15T12:23:14.056027Z",
        "updatedAt": "2019-04-15T12:23:14.056087Z"
    }
}
```
 #### Relevant Pivotal Tracker story
[#164798295](https://www.pivotaltracker.com/story/show/164798295)
#### Screen shots
Screenshot 2019-04-15 at 17 26 30

### CODE REVIEW

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -133,3 +137,116 @@ def post(self, request, **kwargs):
...
+        """
+        slug = kwargs.get('slug')
+        comment = request.data
+        
+        article = get_object_or_404(Articles, slug=slug)
+        author = Profile.objects.get(user_id=request.user.id)
+        # data_to_save = dict(
+        #     body=comment['body']
+        # )
+
```
Comment (e-ian) [inline authors/apps/articles/views.py]: @patrickf949 , I think you can do away with the commented code.

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -133,3 +137,116 @@ def post(self, request, **kwargs):
...
+    pass
+
+
+class ReplyCreateList(generics.CreateAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = serializers.CommentReplySerializer
+    queryset = Reply.objects.all()
+
+
+    def post(self, request, **kwargs):
```
Comment (e-ian) [inline authors/apps/articles/views.py]: Add a doc string for this method. Helps in code readability

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -133,3 +137,116 @@ def post(self, request, **kwargs):
...
+            comment_id=(get_object_or_404(Comment, id=comment_id)).pk
+        )
+        serializer = self.serializer_class(replies, many=True)
+        return Response({
+            'replies': serializer.data
+        },
+            status=status.HTTP_200_OK
+        )
+
+class CommentRetrieveUpdateDestroy(generics.RetrieveUpdateDestroyAPIView):
```
Comment (e-ian) [inline authors/apps/articles/views.py]: @patrickf949 are we using this class?

**Review by e-ian — COMMENTED** [review]
@patrickf949 I have a few comments on your PR

File: authors/apps/articles/views.py (hunk 3 above)
Comment (patrickf949) [reply authors/apps/articles/views.py]: Yes, still a work in progress

**Review by Genza999 — APPROVED** [review]
Great work Fitz, kindly work on the coveralls percentage though.

**Review by e-ian — APPROVED** [review]
Tested. LGTM

File: authors/apps/articles/models.py [hunk 4]
```diff
@@ -64,3 +67,32 @@ class ArticleRating(models.Model):
...
     ratings = models.IntegerField()
     created_at = models.DateTimeField(auto_now_add=True)
+
+
+class Comment(models.Model):
+    """
+    Model for comments
+    """
+    body = models.TextField()
+    parentId = models.IntegerField()
```
Comment (inno-asiimwe) [inline authors/apps/articles/models.py:115]: This would have been better as a foreign key referencing the same model.

**Review by inno-asiimwe — APPROVED** [review]
LGTM

**Pull request merged**